pkg/steps/template.go

```diff
@@ -48,6 +57,52 @@ type templateExecutionStep struct {
 	subTests []*junit.TestCase
 }
 
+func injectResourcesToTemplatePods(template *templateapi.Template, resources api.ResourceConfiguration) error {
+	if resources != nil {
+		containerResources, err := resourcesFor(resources.RequirementsForStep(template.Name))
+		if err != nil {
+			return fmt.Errorf("unable to calculate resources for %s: %s", template.Name, err)
+		}
+
+		for i, object := range template.Objects {
+			var pod *coreapi.Pod
+			if err := yaml.Unmarshal(object.Raw, &pod); err != nil {
+				return fmt.Errorf("couldn't unmashal pod %s: %v", pod.Name, err)
+			}
+
+			if pod != nil {
+				for index, container := range pod.Spec.Containers {
+					if container.Name == TestContainerName {
+						pod.Spec.Containers[index].Resources = containerResources
+						break
+					}
+				}
+
+				corev1Scheme := runtime.NewScheme()
+				utilruntime.Must(coreapi.AddToScheme(corev1Scheme))
+				corev1Codec := serializer.NewCodecFactory(corev1Scheme).LegacyCodec(coreapi.SchemeGroupVersion)
+
+				template.Objects[i] = runtime.RawExtension{
+					Raw:    []byte(runtime.EncodeOrDie(corev1Codec, pod)),
+					Object: pod.DeepCopyObject(),
```

Comment:
Are we doing a deep copy to not change the original? Wouldn't we want to do the deep copy before we change resources?

Reply:
well, this `DeepCopyObject` is just to get the pod as an object for the `RawExtension`.  This pod is the result of the `UnMarsall`  so technically we don't need to `DeepCopy` before changing the resources. So, in the end, we do the `DeepCopy` to actually change the original.